packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Models/Types/SystemModelSerializationTypeProvider.cs

```diff
@@ -0,0 +1,231 @@
+            );
+        }
+
+        /// <summary>
+        /// Builds the I model write method.
+        /// <param name="iPersistableModelTInterface">The <see cref="CSharpType"/> that represents the IPersistableModelT interface.</param>.
+        /// </summary>
+        internal static CSharpMethod BuildIModelWriteMethod(CSharpType iPersistableModelTInterface)
+        {
+            // BinaryData IPersistableModel<T>.Write(ModelReaderWriterOptions options)
+            var returnType = typeof(BinaryData);
+            return new CSharpMethod
+            (
+                new MethodSignature(nameof(IPersistableModel<object>.Write), null, null, MethodSignatureModifiers.None, returnType, null, new[] { SerializationOptionsParameter }, ExplicitInterface: iPersistableModelTInterface),
+                // TO-DO: Call the base model ctor for now until the model properties are serialized https://github.com/microsoft/typespec/issues/3330
+                Snippets.Return(new NewInstanceExpression(returnType, new ValueExpression[] { new StringLiteralExpression(iPersistableModelTInterface.Name, false) })),
+                new CSharpMethodKinds(SystemCSharpMethodKinds.IModelSerializationWrite)
+            );
+        }
+
+        /// <summary>
+        /// Builds the I model create method.
+        /// <param name="iPersistableModelTInterface">The <see cref="CSharpType"/> that represents the IPersistableModelT interface.</param>.
+        /// </summary>
+        internal static CSharpMethod BuildIModelCreateMethod(CSharpType iPersistableModelTInterface)
+        {
+            Parameter dataParameter = new("data", null, typeof(BinaryData), null, ValidationType.None, null);
+            // IPersistableModel<T>.Create(BinaryData data, ModelReaderWriterOptions options)
+            var typeOfT = GetModelArgumentType(iPersistableModelTInterface);
+            return new CSharpMethod
+            (
+              new MethodSignature(nameof(IPersistableModel<object>.Create), null, null, MethodSignatureModifiers.None, typeOfT, null, new[] { dataParameter, SerializationOptionsParameter }, ExplicitInterface: iPersistableModelTInterface),
+              // TO-DO: Call the base model ctor for now until the model properties are serialized https://github.com/microsoft/typespec/issues/3330
+              Snippets.Return(new NewInstanceExpression(typeOfT, Array.Empty<ValueExpression>())),
+              new CSharpMethodKinds(SystemCSharpMethodKinds.IModelDeserializationCreate)
+            );
+        }
+
+        /// <summary>
+        /// Builds the I model GetFormatFromOptions method.
+        /// <param name="iPersistableModelTInterface">The <see cref="CSharpType"/> that represents the IPersistableModelT interface.</param>.
+        /// </summary>
+        internal static CSharpMethod BuildIModelGetFormatFromOptionsMethod(CSharpType iPersistableModelTInterface)
+        {
+            ValueExpression jsonWireFormat = Snippets.Literal("J");
```

Comment:
maybe we could move this to a Serialization Snippet class such as `Snippets.Serializations.JsonFormat` to make it have more semantics